blockchain/substrate/types.go

```diff
@@ -154,6 +154,87 @@ type RoundData struct {
 	Answered_In_Round RoundID
 }
 
+// SubstrateRequestParams allows for decoding a scale hex string into
+// a byte array, which is then encoded back to a scale encoded byte array,
+// to be decoded into a string array. This solves issues where decoding
+// directly into a string array would read past the end of the array.
+type SubstrateRequestParams []string
+
+func (a *SubstrateRequestParams) Decode(decoder scale.Decoder) error {
+	// Decode hex string into a byte array.
+	// This allows us to stop reading where the
+	// intended byte array stops.
+	var bz types.Bytes
+	err := decoder.Decode(&bz)
+	if err != nil {
+		return err
+	}
+
+	// Encode byte array into a scale encoded byte array
+	encoded, err := types.EncodeToBytes(bz)
+	if err != nil {
+		return err
+	}
+
+	// Decode byte array into a string array
+	var strings []string
+	err = types.DecodeFromBytes(encoded, &strings)
+	if err != nil {
+		return err
+	}
+
+	*a = strings
+
+	return nil
+}
+
+func (a SubstrateRequestParams) Encode(_ scale.Encoder) error {
+	return nil
+}
+
+// EventChainlinkOracleRequest is the event structure we expect
+// to be emitted from the Chainlink pallet
+type EventChainlinkOracleRequest struct {
```

Comment:
Maybe the `Chainlink` added in these types is not necessary? Or do we want to distinguish them from other types?

Reply:
The name is there just to work better with the Rust implementation of the pallet. There's quite a few bad "Go styling errors" here, but it'll be necessary for working with the Rust pallet.